### ISSUE TITLE
Performance question

### ISSUE THREAD

**Issue by tszpinda** [dfahlander/Dexie.js#62]
Hi,
Am not sure if am doing anything stupid, but I see quite a bit of diff in performance while comparing these two:
https://gist.github.com/anonymous/18701cafb01b270488e0
https://gist.github.com/anonymous/240a8a4ddb97e8d9ed81

while on the desktop browser the difference is about 1sec, on mobile its more the 1 min (5sec vs 85sec) have you got any ideas? maybe be you some benchmarks etc? (mobile - using android chrome browser, on default android browser its slow too).

**Comment by tszpinda** [comment]
adding transaction around it for Dexie helps a bit:

https://gist.github.com/anonymous/f37c9e6a95ceda2a529d

down to 22sec on mobile

**Comment by tszpinda** [comment]
looking at the chrome tools lots of time is wasted by this:
    function fakeAutoComplete(fn) {
        var to = setTimeout(fn, 1000);
        clearTimeout(to);
    }

can this be changed/removed/improved?

**Comment by dfahlander** [comment]
When coding with visual studio, those lines are priceless but we should have a build script that removes them in the release / minified version. I did a simple profiling of that sequence last year but came to the conclusion that they really fast in the particular browser i tested it in (probably some version of chrome, opera or ie) but did not dig so deep into it.

There is a general problem that Dexie doesn't have a build script that I hope that someone could help or with.

**Comment by thebestclass** [comment]
Any solution? this delay breaks my code. @dfahlander

**Comment by dfahlander** [comment]
That piece of code is not needed and does nothing. It's just a way to improve the IDE 's code completion. Once a build script is in place (grunt, gulp or so), i'll remove those calls from the minified version. 

What you can do right now, is either to remove the code in fakeautocomplete(), or to remove all calls to it. Will not affect behavior at all. Sorry I wasn't more clear in my last answer about the actual code change you could do.

**Comment by dfahlander** [comment]
The calls to setTimeout()/clearTimeout() are now removed. However, when testing performance with chrome/opera/ie/firefox, this change is not noticable. Haven't tested on safari though.

**Comment by tszpinda** [comment]
David, the difference for me was noticeable on mobile devices.

**Comment by dfahlander** [comment]
Mobile devices is a common use case for the library so hopefully this fix might improve performance a little then.

Thanks!
David

**Comment by yazarloo** [comment]
Thank you dear **David Fahlander**.
I found Dexie as a best solution for storing and recovering data at the client side. I implemented a solution with mixing the Dexie.js and Linq.js. It's amazing :) . But I think there is a bottleneck. When I want to select all data of a table with around 100k records and converting it to array there is a delay about 5 second. That is an issue that I don't know how to overcome that.
Best regards.
